block/manager.go

```diff
@@ -184,6 +188,24 @@ func NewManager(
 		return nil, err
 	}
 
+	if instruction, forkNeeded := m.forkNeeded(); forkNeeded {
+		// Set proposer to nil
+		m.State.SetProposer(nil)
+
+		// Upgrade revision on state
+		state := m.State
+		state.Version.Consensus.App = instruction.Revision
+		state.RevisionStartHeight = instruction.RevisionStartHeight
+		if instruction.RevisionStartHeight == m.State.NextHeight() {
+			drsVersion, err := strconv.ParseUint(version.DrsVersion, 10, 32)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse drs version")
+			}
+			state.RollappParams.DrsVersion = uint32(drsVersion)
```

Comment:
this needs to be changed to enable the rollapp param validation to pass (which is also done when starting the node) when its necessary to start with the new drs version.  conversion is necessary because, afaik, its only possible to set ldflags using strings. i added conversion func in the same version package.